src/main/java/com/google/codeu/servlets/UserListServlet.java

```diff
@@ -0,0 +1,22 @@
+package com.google.codeu.servlets;
+
+import java.io.IOException;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Handles fetching all users for the community page.
+ */
+@WebServlet("/user-list")
+public class UserListServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+
+    response.getOutputStream().println("this will be my user list");
```

Comment:
Please add a WIP (work in progress) comment (or, even better, add a [WIP] tag to the PR title). This is because (in my opinion) it's good hygiene for master to be in a state that is acceptable to deploy to production users. If this were merged to master right now users might see testing code if we deployed.

To add changes on top of this change, you can set future PRs to target this branch. (i.e. once you write the HTML, create a PR from CommunityPageHTML -> CommunityPage). Then we can review the second PR independently. To add more changes, target the CommunityPageHTML branch, etc. And when they're all approved we can merge them all.

Another strategy you could use is make your servlet return a 404 Not Found instead for now. That way, you can get your code checked in to master, but users won't see any disruption/in progress code.

Reply:
By this comment I meant — add a WIP tag and *don’t* merge into master. We can merge after you finish the functionality in other PRs.

Reply:
ok my apologies i misunderstood! Is there any way to reverse it, if not what should I do? @tylerhou

Reply:
No worries; it’s not a big deal. Just a heads up for next time.

Reply:
We’re not actually serving production traffic so ¯\_(ツ)_/¯

Reply:
ok sounds good! so for my next section I have to make a community HTML page, how should I do this differently?

Reply:
There are two ways:

## Chaining PRs

For your next PR you can target `master`. If it's a work in progress (i.e. has code that production users shouldn't see), then you can get the necessary approvals -- just remember not to submit. Let's call that PR `Feature1`.

So you have `Feature1` done, with a PR to master. Then you can create a branch `Feature2`. When you're ready for `Feature2` to be shipped, then you can create a new PR which tries to merge `Feature2` into `Feature1`.  See step 4 in [Creating the Pull Request](https://help.github.com/en/articles/creating-a-pull-request#creating-the-pull-request). 

If you need to create another change on top of `Feature1` and `Feature2`, then you can create a new branch `Feature3` and do the above to merge `Feature3` into `Feature2`.

This creates a stack of changes: `Feature3` -> `Feature2` -> `Feature1` -> `master`.

Once all your PRs are approved, we can merge them all into master, collapsing from the top: Merge `Feature3` -> `Feature2`; then merge `Feature2 (with Feature3) ` -> `Feature1`, then merge `Feature1 (with Feature2 and Feature3)` -> `master`.

## Merge into master, but don't disrupt production users

If you don't want to manage this PR chain, and you want to get your code into master right away (maybe someone else needs your code for their feature) you can also edit the PR so that it doesn't have any user-facing changes. In this case, that means modifying the `UserListServlet` to return 404. That way you can get your backend/data model changes checked in without risking users seeing test/in progress code.